integration_tests/e2e/confirm-achieve-goal.cy.ts

```diff
@@ -0,0 +1,152 @@
+          cy.wrap(goal).as('newGoal')
+          cy.addStepToGoal(goal.uuid, DataGenerator.generateStep({}))
+        })
+
+        planOverview.agreePlan()
+        cy.get<Goal>('@newGoal').then(goal => {
+          cy.visit(`/update-goal-steps/${goal.uuid}`)
+
+          cy.get('#step-status-1').select('Completed') // select completed status
+
+          cy.get('button').contains('Save goal and steps').click()
+        })
+      })
+    })
+
+    it('Confirm goal achieved successfully without optional note', () => {
+      achieveGoal.isGoalAchievedRadio('yes')
+      cy.get('button').contains('Save and continue').click()
+      cy.url().should('include', 'plan?type=achieved&status=achieved')
+      cy.get(':nth-child(3) > .moj-sub-navigation__link')
+      cy.get('.moj-sub-navigation__link').eq(2).should('contain', 'Achieved goals (1)')
+      cy.get('.govuk-summary-card').should('contain', 'Marked as achieved on')
+
+      cy.get<Goal>('@newGoal').then(goal => {
+        cy.visit(`/view-achieved-goal/${goal.uuid}`)
+      })
+      cy.get('.govuk-body').should('contain', 'Marked as achieved on')
+      cy.checkAccessibility()
+    })
+
+    it('Confirm goal achieved successfully with optional note', () => {
+      achieveGoal.isGoalAchievedRadio('yes')
+      cy.get('#goal-achievement-helped').type('Some optional text in the achievement note field')
+      cy.get('button').contains('Save and continue').click()
+      cy.url().should('include', 'plan?type=achieved&status=achieved')
+      cy.get(':nth-child(3) > .moj-sub-navigation__link')
+      cy.get('.moj-sub-navigation__link').eq(2).should('contain', 'Achieved goals (1)')
+
+      cy.get('.moj-sub-navigation__link').eq(2).click()
+      planOverview.getSummaryCard(0).get('a').contains('View details').click()
+      cy.get('.govuk-details__text').should('contain', 'Some optional text in the achievement note field')
+      cy.get('p.govuk-body').should('contain', 'Marked as achieved on ')
+      cy.get('a.govuk-back-link').should('have.attr', 'href').and('include', '/plan?type=achieved')
+      cy.get('.govuk-button--secondary').should('have.length', 0)
+      cy.checkAccessibility()
+    })
+
+    it('Confirm errors are displayed with optional note of more than 4000 characters', () => {
+      const lorem = faker.lorem.paragraphs(40)
+      achieveGoal.isGoalAchievedRadio('yes')
+      cy.get('#goal-achievement-helped').invoke('val', lorem)
+      cy.get('button').contains('Save and continue').click()
+      cy.url().should('include', '/confirm-if-achieved/')
+      cy.get('.govuk-error-summary').should(
+        'contain',
+        'How achieving this goal has helped must be 4,000 characters or less',
+      )
+      cy.get('#goal-achievement-helped-error').should(
+        'contain',
+        'How achieving this goal has helped must be 4,000 characters or less',
+      )
+      cy.get('#goal-achievement-helped').should('not.be.empty')
```

Comment:
isn't this always true if line 122 has passed?

Reply:
This is checking that the data in the textbox still exists when we have a validation error. To make this a bit clearer I have changed it to check against the content rather than just whether it is not empty